testing/test-helper.js

```diff
@@ -45,6 +45,27 @@ export function stubServiceForDoc(sandbox, ampdoc, serviceId, method) {
 }
 
 /**
+ * Resolves a promise when the method stub is called for the first time.
+ */
+export function stubWithCalledPromise(sandbox, object, method, opt_callMethod) {
+  const original = object[method];
+  let stub;
+  const promise = new Promise(resolve => {
+    stub = sandbox.stub();
+    const spy = sandbox.stub(object, method, () => {
+      const result = stub.apply(object, spy.lastCall.args);
```

Comment:
`object` here could be the incorrect context. I think `spy.lastCall` has a `context` property?